shared/chat/conversation/container.js

```diff
@@ -1,27 +1,66 @@
 // @flow
 import * as React from 'react'
 import * as Constants from '../../constants/chat2'
+import * as Chat2Gen from '../../actions/chat2-gen'
 import * as Types from '../../constants/types/chat2'
 import * as Flow from '../../util/flow'
 import {isMobile} from '../../styles'
-import {connect} from '../../util/container'
+import {connect, getRouteProps} from '../../util/container'
 import Normal from './normal/container'
 import NoConversation from './no-conversation'
 import Error from './error/container'
 import YouAreReset from './you-are-reset'
 import Rekey from './rekey/container'
+import flags from '../../util/feature-flags'
 
 type OwnProps = {||}
 
 type SwitchProps = {
   conversationIDKey: Types.ConversationIDKey,
   isPending: boolean,
+  isFocused?: boolean,
+  selectConversation: () => void,
+  deselectConversation: () => void,
   type: 'error' | 'noConvo' | 'rekey' | 'youAreReset' | 'normal' | 'rekey',
 }
 
 const DONT_RENDER_CONVERSATION = __DEV__ && false
 
+let NavigationEvents
+if (flags.useNewRouter) {
+  NavigationEvents = require('@react-navigation/core').NavigationEvents
+} else {
+  NavigationEvents = () => null
+}
+
 class Conversation extends React.PureComponent<SwitchProps> {
+  _onDidFocus = () => {
```

Comment:
convo now has to tell the system when it's the visible conversation so mark as read work correctly on tab switches